sdk/textanalytics/Azure.AI.TextAnalytics/tests/Infrastructure/RetryOnInternalServerErrorAttribute.cs

```diff
@@ -26,10 +25,29 @@ public RetryOnInternalServerErrorAttribute()
         /// </summary>
         private static bool ShouldRetry(TestExecutionContext context)
         {
-            return
-                context.CurrentResult.Message.Contains("Azure.RequestFailedException")
-                && context.CurrentResult.Message.Contains("Status: 200 (OK)")
-                && context.CurrentResult.Message.Contains("ErrorCode: InternalServerError");
+            string message = context?.CurrentResult?.Message;
```

Comment:
Curious: is there anything here that would be worth rolling into a test-level attribute, so other libraries could piggy-back on the configurability?

Reply:
Yes! This one is actually a test-level attribute by virtue of inheriting from the [`RetryOnErrorAttribute`](https://github.com/Azure/azure-sdk-for-net/blob/0c2c7da33c55fe68b9f11ebc45ac0defd2d56eaa/sdk/core/Azure.Core.TestFramework/src/RetryOnErrorAttribute.cs) that I put in Azure.Core.TestFramework. Other libraries can create their own sub-class and tailor it to their needs.